src/status_im2/contexts/wallet/home/view.cljs

```diff
@@ -38,41 +38,53 @@
    {:id :collectibles :label (i18n/label :t/collectibles) :accessibility-label :collectibles-tab}
    {:id :activity :label (i18n/label :t/activity) :accessibility-label :activity-tab}])
 
+(def aval
+  (atom {:a 1
+         :b 2}))
+
+(reset! aval {:a 4})
+
+(defn f-view-internal
+  [selected-tab]
+  (let [networks           (rf/sub [:wallet/network-details])
+        account-cards-data (rf/sub [:wallet/account-cards-data])
+        accounts           (rf/sub [:wallet/accounts])
+        cards              (conj account-cards-data (new-account-card-data))
+        top                (safe-area/get-top)]
+    (rn/use-effect (fn []
+                     (rf/dispatch [:wallet/request-collectibles
+                                   {:start-at-index 0
+                                    :new-request?   true}]))
+                   [(count accounts)])
```

Comment:
@J-Son89, @ulisesmac, @briansztamfater, you may be interested:

Here I use `use-effect` because when user goes straight to wallet page, there is a moment when details about accounts aren't returned from `status-go`. And I've found kind of a problem: if you will try to use `[accounts]`, or `[@accoutns]` - hook won't work properly and will call its function on every re-render. But with `[(count accounts)]` it works. So it looks like only simple values can be used as a conditional launch of `use-effect` hook.
I tried with atoms, reagent atom, clojure maps... all these kinds of values stated as condition trigger hook on every re-render( 
If you know something about this behaviour - please share.

Reply:
> @J-Son89, @ulisesmac, @briansztamfater, you may be interested:
> 
> Here I use `use-effect` because when user goes straight to wallet page, there is a moment when details about accounts aren't returned from `status-go`. And I've found kind of a problem: if you will try to use `[accounts]`, or `[@accoutns]` - hook won't work properly and will call its function on every re-render. But with `[(count accounts)]` it works. So it looks like only simple values can be used as a conditional launch of `use-effect` hook.
> I tried with atoms, reagent atom, clojure maps... all these kinds of values stated as condition trigger hook on every re-render( 
> If you know something about this behaviour - please share.

Hey @vkjr

I have faced this same issue in the past.
The problem is that the hook dependencies are being compared in the JS side to check if the hook should be re-executed.

It means that in JS, numbers can be compared without any problem, (e.g. `4` = `4` always is true), but JS objects are compared by reference, so two objects that look exactly the same are different
 (e.g.
 `#js{"0x123" #js{"a" 100}}`
=
`#js{"0x123" #js{"a" 100}}`
Always is false unless you store it in a `let` binding and compare the same var)

As far as I remember, the `use-effect` hook we are using transforms the dependencies vector from CLJS to JS, so it's creating new objects from clojure maps for each re-render.

Ok, enough of explanations. Let's give a solution.

We can rely on strings :)  since they are compared by value.

So if we have a clojure map for accounts that looks like:

```clojure
{"0x12..." {...}
 "0x01..." {...}
 "0xAB..." {...}}
```

I'd do:

```clojure
(-> @accounts keys sort)
```

So that we get an ordered list of accounts that also works as hook dependency, if an account is added, the effect will detect a change and will re render, and I'm using `sort` because clojure hash maps aren't ordered (unless `sorted-map` is used), and we don't want to trigger renders just because of the order changed.

Note: maybe we'd need to add `vec` at the end of that `->` if a vector must be passed to our use-effect dependency.